### PR TITLE
feat: application flow

### PR DESCRIPTION
**Pull request by shaohuzhang1**
feat: application flow

### CODE REVIEW

File: apps/application/flow/tools.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+
+def to_stream_response_simple(stream_event):
+    r = StreamingHttpResponse(
+        streaming_content=stream_event,
+        content_type='text/event-stream;charset=utf-8',
+        charset='utf-8')
+
+    r['Cache-Control'] = 'no-cache'
+    return r
```
Comment (shaohuzhang1) [inline apps/application/flow/tools.py:191]: ### Code Review

#### Issues Identified:

1. **Line Length**: The file exceeds the recommended 80-character line length standard, making it harder to read.

2. **Comments**: There are many comments that could potentially be removed as they may no longer be relevant or needed given the current context.

3. **Docstrings**: While some docstrings exist, they need more specific details about the purpose of each function or what kind of parameters they accept.

4. **Function Redundancies**: 
    - `to_response` and `to_response_simple` have similar logic but serve different purposes (service vs. simple service). Consider rethinking this pattern if possible.
  
5. **Error Handling in Streams**:
    ```python
    try:
        [your code here]
    except Exception as e:
        error_message = str(e)
        # Write context and handler with the error message...
    ```
    This can cause issues with streams, as exceptions within a stream generator might break the flow prematurely due to buffering or network constraints which isn't desired.

6. **Variable Names**: Some variable names like `answer`, `chunk`, `reasoning_content_chunk` could be improved for clarity or brevity.

7. **Resource Management**: Ensure that resources (like HTTP connections) are properly closed after use to prevent resource leaks.

8. **Code Duplication**: There's some redundancy between how responses are processed and formatted into JSON objects.

#### Optimization Suggestions:

1. **Increase Readability**:
    - Break long lines by using parentheses inside functions when necessary rather than splitting them across multiple lines.
    
2. **Refactor Logic**:
    - Consider merging `handle_error_in_stream` into a single function with appropriate parameters so that all parts of your stream processing logic remain coherent and easier to maintain.

3. **Improve Error Reporting**:
    - Provide more detailed error reporting by capturing stack traces or including additional information within the response payload. 

4. **Utilize Built-in Stream Processing Libraries**:
    - If applicable, consider using Python’s built-in streaming libraries (`asyncio.EventLoop`) for handling asynchronous streaming operations efficiently.

5. **Test Thoroughly**:
    - Validate the behavior under various scenarios, especially focusing on edge cases like empty requests or errors during generation processes.

By addressing these points, you will enhance the readability, robustness, and efficiency of your codebase while improving user experience related to API calls involving streaming responses.
